src/mux_client_impl/command.rs

```diff
@@ -0,0 +1,103 @@
+use super::Error;
+use super::RemoteChild;
+use super::{as_raw_fd, ChildStderr, ChildStdin, ChildStdout, Stdio};
+
+use std::ffi::OsStr;
+use std::path::PathBuf;
+
+use openssh_mux_client::connection::{Connection, EstablishedSession, Session};
+
+#[derive(Debug)]
+pub struct Command {
+    cmd: String,
+    ctl: PathBuf,
+
+    stdin_v: Stdio,
+    stdout_v: Stdio,
+    stderr_v: Stdio,
+}
+
+impl Command {
+    pub(crate) fn new(ctl: PathBuf, cmd: String) -> Self {
+        Self {
+            cmd,
+            ctl,
+
+            stdin_v: Stdio::null(),
+            stdout_v: Stdio::null(),
+            stderr_v: Stdio::null(),
+        }
+    }
+}
+
+impl Command {
+    pub fn arg<S: AsRef<str>>(&mut self, arg: S) -> &mut Self {
+        self.cmd.push_str(" '");
+        self.cmd.push_str(arg.as_ref());
+        self.cmd.push('\'');
+        self
+    }
+
+    pub fn raw_arg<S: AsRef<OsStr>>(&mut self, arg: S) -> &mut Self {
+        self.cmd.push(' ');
+        self.cmd.push_str(&arg.as_ref().to_string_lossy());
+        self
+    }
+
+    pub fn stdin<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
+        self.stdin_v = cfg.into();
+        self
+    }
+
+    pub fn stdout<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
+        self.stdout_v = cfg.into();
+        self
+    }
+
+    pub fn stderr<T: Into<Stdio>>(&mut self, cfg: T) -> &mut Self {
+        self.stderr_v = cfg.into();
+        self
+    }
+
+    /// After this function, stdin, stdout and stderr is reset.
+    async fn spawn_impl(
```

Comment:
Any particular reason you have this be a separate function rather than just directly in `spawn`?

Reply:
It was added during development of `mux_client_impl`.
I thought I was going to add some complicated crazy branches, but turns out it was just fine.

So it is actually not useful anymore.

Reply:
Removed.